### ISSUE TITLE
Docker Engine Not Starting

### ISSUE THREAD

**Issue by treiter317** [docker/for-win#13547]
### Description

The docker engine refuses to start on Windows 11 PC and shows loading spinner infinitely. Previously tried to run diagnostics test when using WSL2 but that failed, so did manual diagnostics from terminal and discovered several failing tests but was unable to fix. Now, using Hyper V, the engine instead says "failed to start" and diagnostics in the docker desktop succeeded, which is the code attached to this post. Have spent a lot of time trying to fix this and no solution online seem to work.

### Reproduce

1. Open docker desktop
2. Either start engine spinner will run infinitely or engine will fail to start

### Expected behavior

I would expect the docker engine to work the first time I downloaded it :( .

### docker version

```bash
error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/version": open //./pipe/docker_engine: The system cannot find the file specified.
Client:
 Cloud integration: v1.0.33
 Version:           24.0.2
 API version:       1.43
 Go version:        go1.20.4
 Git commit:        cb74dfc
 Built:             Thu May 25 21:53:15 2023
 OS/Arch:           windows/amd64
 Context:           default
```

### docker info

```bash
Client:
 Version:    24.0.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.5
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.18.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.19
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scan.exe
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  v0.12.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
ERROR: error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/info": open //./pipe/docker_engine: The system cannot find the file specified.
errors pretty printing info
```

### Diagnostics ID

84BF44E0-88A4-4F12-897F-0265E89C4312/20230615155721

**Comment by grosssmutttr** [comment]
Same here.
Upgraded today to version 4.20.1 and now it is stuck on loading forever.

I tried different approaches:
1. Restart PC
2. Restart wsl
3. Update wsl
4. Tried wsl --unregister docker-desktop wsl and --unregister docker-desktop-data
5. ....

And if I try to turn on / activate "Enable integration with additional distros" for Ubuntu I'm getting a Windows blue screen and the PC restarts...

**Comment by TristisOris** [comment]
same begins with 4.18, after win update probably. exist with 4.20.

```
System.Exception:
1 error occurred:
	* waiting for docker daemon: context deadline exceeded


   at Docker.Engines.WSL2.LinuxWSL2Engine.<DoStartAsync>d__10.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.Engines\WSL2\LinuxWSL2Engine.cs:line 93
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__7.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 79
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__7.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.Engines.Engines.<WaitCurrentTransitionsAsync>d__30.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.Engines\Engines.cs:line 290
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.Engines.<StartAsync>d__24.MoveNext() in C:\workspaces\PR-22253\src\github.com\docker\pinata\win\src\Docker.Engines\Engines.cs:line 114
```

**Comment by zakjanzi** [comment]
Same. I am unable to Clean / Purge data from the Troubleshoot section, and resetting to factory settings (and checking the  "delete WSL and Hyper V" boxes) also did not work. (it wouldn't reset).

This happened after launching the docker desktop app right after installing. (I also tried restarting along with a few clean installs)

**Comment by sambasig** [comment]
i am also facing same issue.Do you got any solution for this?

**Comment by treiter317** [comment]
I was able to resolve the issue, but I ended up having to use Hyper V instead of WSL2.

Try the top answer here:
https://stackoverflow.com/questions/61289669/docker-desktop-not-starting-sequence-contains-no-matching-element

**Comment by lorenrh** [comment]
We have released Docker desktop 4.21.1, containing a fix for this issue, please see https://docs.docker.com/desktop/release-notes/

**Comment by sambasig** [comment]
Thank you
________________________________
